clients/client-s3/test/unit/dotSegmentInUriLabel.spec.ts

```diff
@@ -0,0 +1,72 @@
+/// <reference types="mocha" />
+import { HttpHandler, HttpRequest, HttpResponse } from "@smithy/protocol-http";
+import { HttpHandlerOptions } from "@smithy/types";
+import { S3 } from "../../src/S3";
+import chai from "chai";
+import chaiAsPromised from "chai-as-promised";
+
+chai.use(chaiAsPromised);
+const { expect } = chai;
+
+/**
+ * Throws an expected exception that contains the serialized request.
+ */
+class EXPECTED_REQUEST_SERIALIZATION_ERROR extends Error {
```

Comment:
this is a constant naming convention, which I suppose is copied over from elsewhere, but it should probably be `ExpectedRequestSerializationError`.

Reply:
Yup. It's copied from protocoltests attempted in https://github.com/aws/aws-sdk-js-v3/pull/6178
Changed to `ExpectedRequestSerializationError` in [`14ca1ec` (#6262)](https://github.com/aws/aws-sdk-js-v3/pull/6262/commits/14ca1ec4897cfbf7886e711aacf78f72610eac67)